etcdserver/auth/auth.go

```diff
@@ -281,11 +285,14 @@ func (s *store) UpdateUser(user User) (User, error) {
 		return old, err
 	}
 
-	hash, err := s.HashPassword(user.Password)
-	if err != nil {
-		return old, err
+	if user.Password != "" {
+		var hash string
+		hash, err = s.HashPassword(user.Password)
```

Comment:
I mean a few line under this, there is a merge call. In the merge call, the passed in user will be merged with the old user. If the old user has password, but the new user does not. The old password should be merged into new user. No?